<!DOCTYPE html>
<html lang="en">
    
    


    <head>
    <link href="https://gmpg.org/xfn/11" rel="profile">
    <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<!-- Enable responsiveness on mobile devices -->
<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
<meta name="generator" content="Hugo 0.105.0">

    
    
    

<title>Code &amp; Resources • Albert Vill</title>


<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Code &amp; Resources"/>
<meta name="twitter:description" content="Expand details…minCED (Mining CRISPRs in Environmental Datasets) is a program used by prokka and other pipelines for the annotation of CRISPRs in metagenomic assemblies. However, manually parsing the output of minCED for interesting CRISPRs is tedious, and comparing CRISPRs across samples is difficult because of differences in array assembly and orientation. CRISPRviewR attempts to address this issue by associating CRISPR arrays by shared consensus repeat sequences, permitting clustering and visualization of spacers between metagenomic assemblies."/>

<meta property="og:title" content="Code &amp; Resources" />
<meta property="og:description" content="Expand details…minCED (Mining CRISPRs in Environmental Datasets) is a program used by prokka and other pipelines for the annotation of CRISPRs in metagenomic assemblies. However, manually parsing the output of minCED for interesting CRISPRs is tedious, and comparing CRISPRs across samples is difficult because of differences in array assembly and orientation. CRISPRviewR attempts to address this issue by associating CRISPR arrays by shared consensus repeat sequences, permitting clustering and visualization of spacers between metagenomic assemblies." />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://acvill.github.io/projects/" /><meta property="article:section" content="" />

<meta property="og:site_name" content="Title" />



    


<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">








<link rel="stylesheet" href="/scss/hyde-hyde.83a929a933962c7643b655554c2e1b7cb32e23c25a2d7262666d4ac0574b9e7e.css" integrity="sha256-g6kpqTOWLHZDtlVVTC4bfLMuI8JaLXJiZm1KwFdLnn4=">


<link rel="stylesheet" href="/scss/print.2744dcbf8a0b2e74f8a50e4b34e5f441be7cf93cc7de27029121c6a09f9e77bc.css" integrity="sha256-J0Tcv4oLLnT4pQ5LNOX0Qb58&#43;TzH3icCkSHGoJ&#43;ed7w=" media="print">



    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
    <script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
    <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
    <!-- Icons -->
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/img/apple-touch-icon.png">
    <link rel="shortcut icon" href="/img/favicon-32x32.png">
    
    

    
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"> </script>
</head>


    <body class=" ">
    
<div class="sidebar">
  <div class="container ">
    <div class="sidebar-about">
      <span class="site__title">
        <a href="https://acvill.github.io/">
        
          Albert Vill
        
        </a>
      </span>
      
        
        
        
        <div class="author-image">
          <img src="https://acvill.github.io/img/about/july2021_cropped1.png" alt="Author Image" class="img--circle img--headshot element--center">
        </div>
        
      
      
      
      <p class="site__description">
        
      </p>
    </div>
    <div class="collapsible-menu">
      <input type="checkbox" id="menuToggle">
      <label for="menuToggle">Albert Vill</label>
      <div class="menu-content">
        <div>
	<ul class="sidebar-nav">
		 
		 
			 
				<li>
					<a href="/about">
						<span>About</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/cv">
						<span>CV</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/posts/">
						<span>Blog</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/projects">
						<span>Code &amp; Resources</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/contact">
						<span>Contact Me</span>
					</a>
				</li>
			 
		
	</ul>
</div>

        <section class="social">
  
	<a href="https://biology.stackexchange.com/users/52196/acvill" rel="me"><i class="fab fa-stack-exchange fa-lg" aria-hidden="true"></i></a>
	
	
	<a href="https://twitter.com/_AlbertVill" rel="me"><i class="fab fa-twitter fa-lg" aria-hidden="true"></i></a>
	
	
	
	
	<a href="https://github.com/acvill" rel="me"><i class="fab fa-github fa-lg" aria-hidden="true"></i></a>
	
	
	
	
	
	
	
	<a href="https://linkedin.com/in/albert-vill-a457821a1" rel="me"><i class="fab fa-linkedin fa-lg" aria-hidden="true"></i></a>
	
	
	<a href="https://scholar.google.com/citations?user=j9imsbgAAAAJ" rel="me"><i class="fas fa-graduation-cap fa-lg" aria-hidden="true"></i></a>
	
	
	
	
	
	
	
	
	
	
	
	
</section>



      </div>
    </div>
    
<div class="copyright">
  &copy; 2020 - 2023 A.C. Vill
  
  <br>
    <a href="https://creativecommons.org/licenses/by-sa/4.0">CC BY-SA 4.0</a>
  
</div>



  </div>
</div>

        <div class="content container">
            
    
<article>
  <header>
    <h1>Code &amp; Resources</h1>
    
    
<div class="post__meta">
    
    
    
    
    
    
    <br/>
    
</div>


  </header>
  
  
  <div class="post">
    


<p><a href="https://github.com/acvill/CRISPRviewR"><img src="https://github-readme-stats.vercel.app/api/pin/?username=acvill&amp;repo=CRISPRviewR&amp;bg_color=21436e&amp;text_color=ededed&amp;title_color=30ff53&amp;hide_border=true&amp;icon_color=ededed" /></a></p>
<details>
<summary>
Expand details…
</summary>
<p><a href="https://github.com/ctSkennerton/minced">minCED</a> (<strong>Min</strong>ing <strong>C</strong>RISPRs in <strong>E</strong>nvironmental <strong>D</strong>atasets) is a program used by <a href="https://github.com/tseemann/prokka">prokka</a> and other pipelines for the annotation of CRISPRs in metagenomic assemblies. However, manually parsing the output of minCED for interesting CRISPRs is tedious, and comparing CRISPRs across samples is difficult because of differences in array assembly and orientation. CRISPRviewR attempts to address this issue by associating CRISPR arrays by shared consensus repeat sequences, permitting clustering and visualization of spacers between metagenomic assemblies. This package also includes a function to <a href="https://albertvill.com/posts/crt_repeats/">fix truncated repeats</a>. Check out the <a href="https://albertvill.com/CRISPRviewR-vignette.html">CRISPRviewR vignette</a> for use cases and example plots.</p>
<center>
<div class="figure">
<img src="/img/about/CRISPR23_oralmicrobiomes_numbers_bgwhite.svg" title="CRISPR arrays" style="width:90.0%" alt="" />
<p class="caption">Example output from CRISPRviewR</p>
</div>
</center>
</details>
<p><a href="https://github.com/acvill/bracken_plot"><img src="https://github-readme-stats.vercel.app/api/pin/?username=acvill&amp;repo=bracken_plot&amp;bg_color=21436e&amp;text_color=ededed&amp;title_color=30ff53&amp;hide_border=true&amp;icon_color=ededed" /></a></p>
<details>
<summary>
Expand details…
</summary>
<p>As a graduate student studying metagenomics, I frequently use Derrick Wood’s <a href="https://doi.org/10.1186/s13059-019-1891-0">Kraken2</a> software for accurate and lightning-fast classification of short reads. <a href="https://doi.org/10.7717/peerj-cs.104">Bracken</a> is complimentary software that uses Kraken reports to produce genus- and species-level abundance estimates. Often, a first step in metagenomic analysis is examining the distribution of different organisms across samples. To that end, I’ve created a <a href="https://acvill.shinyapps.io/bracken_plot/">simple shiny app</a> for quick, customizable plotting of taxonomic structures encoded in multi-sample Bracken reports.</p>
<center>
<div class="figure">
<img src="/img/about/bracken_barplot.png" title="Bracken bar plot" style="width:75.0%" alt="" />
<p class="caption">Example output from the bracken_plot app showing genus-level relative abudance across oral microbiome samples. Don’t mind the human read contamination!</p>
</div>
</center>
</details>
<p><a href="https://github.com/acvill/nodeSeqs"><img src="https://github-readme-stats.vercel.app/api/pin/?username=acvill&amp;repo=nodeSeqs&amp;bg_color=21436e&amp;text_color=ededed&amp;title_color=30ff53&amp;hide_border=true&amp;icon_color=ededed" /></a></p>
<details>
<summary>
Expand details…
</summary>
<p><code>nodeSeqs.sh</code> takes in a <a href="http://gfa-spec.github.io/GFA-spec/GFA1.html"><strong>G</strong>raphical <strong>F</strong>ragment <strong>A</strong>ssembly file</a> and outputs sequences proximal to high-degree nodes. In the language of GFA, this script looks for <em>segments</em> that have a large number of <em>links</em>, and then filters those segments by <em>k</em>-mer coverage to preclude hits that are poorly assembled due to low coverage. The resulting .fasta can then be fed into your favorite sequence alignment software.</p>
<p>For an example application of this script, please see my blog post: <a href="https://albertvill.com/posts/nodeseqs_oral/"><strong><em>Lost at the crossroads: genes at the nodes of short-read assembly graphs</em></strong></a>.</p>
<center>
<div class="figure">
<img src="/img/about/bandage_sub.svg" title="Bandage Graph" style="width:90.0%" alt="" />
<p class="caption">A partial Bandage graph for a plasmidome assembly shows segments connected by links</p>
</div>
</center>
</details>
<p><a href="https://github.com/acvill/findMotifs"><img src="https://github-readme-stats.vercel.app/api/pin/?username=acvill&amp;repo=findMotifs&amp;bg_color=21436e&amp;text_color=ededed&amp;title_color=30ff53&amp;hide_border=true&amp;icon_color=ededed" /></a></p>
<details>
<summary>
Expand details…
</summary>
<p>To understand the genetic context and biological relevance of DNA motifs, it’s often valuable to analyze their distribution and sequence conservation. Many programs exist for both the identification of novel motifs and the scoring of known motifs, though few of these tools work out-of-the-box for highly fragmented bacterial genomes. To address this, I’ve written findMotifs.R – an easy-to-implement R script to find and score short sequences using position-frequency matrices (PFMs). Given a set of contigs and a list of PFMs, this script returns an easy-to-parse table containing the sequence, score, and strand-wise position of each match above a user-defined threshold. findMotifs.R requires R version 4.0 and three packages: Biostrings, optparse, and stringr.</p>
<center>
<div class="figure">
<img src="/img/about/findmotifs2.PNG" title="command line screenshot of findMotifs in action" style="width:90.0%" alt="" />
<p class="caption">findmotifs.R is invoked from the command line with user inputs. Progress for each motif prints to the console.</p>
</div>
</center>
</details>
<style>
p.caption {
  font-size: 0.9em;
  text-align: center;
  color: #7f1cad;
  margin-right: 10%;
  margin-left: 10%;
}
details > summary {
  font-size: 1.1em;
  padding: 0px;
  background-color: #FFFFFF;
  border: none;
  box-shadow: 0px 0px 0px #bbbbbb;
  cursor: pointer;
  margin-left: 5%;
  color: #21436e;
  border-radius: 5px;
}
</style>

  </div>
  


  

  
    


</article>


        </div>
        
    

  
  
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'G-JKQWQF4NBY', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>


<script defer src="https://use.fontawesome.com/releases/v5.12.1/js/all.js" integrity="sha384-ZbbbT1gw3joYkKRqh0kWyRp32UAvdqkpbLedQJSlnI8iLQcFVxaGyrOgOJiDQTTR" crossorigin="anonymous"></script>


    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.13.1/highlight.min.js"></script>
        
    <script type="text/javascript">
        
        hljs.initHighlightingOnLoad();
    </script>
    



    



    </body>
</html>
