---
title: "About me"
author: "Albert Vill"
date: "2 May 2022"
output: html_document
---

<center>

![A waterfall in Connecticut Hill Wildlife Area, Newfield, NY](/img/about/waterfall2.jpg)

</center>

## :pizza::petri_dish: Background 

I am from New Jersey. New Yorkers may think they have better bagels and pizza, but they're wrong. 

I graduated from Gettysburg College in 2016 with a B.S. in biochemistry and molecular biology. As a first-generation college student, I got my initial exposure to scientific research in the [PhagesRock lab](https://xsigsummer.wordpress.com/?s=phage) under the mentorship of professors VÃ©ronique Delesalle and Greg Krukonis. 

After undergrad, I enrolled at Cornell University as a graduate student in the field of Genetics, Genomics & Development, and shortly thereafter joined [the Brito Lab](https://www.britolab.org) in the Meinig School of Biomedical Engineering. My graduate work involved the application of different sequencing techniques to understand the DNA and RNA content of the human gut microbiome.

I defended my thesis in April 2022, titled *Metagenomic Methods to Investigate Mobile Element Context and Nascent Transcription in the Human Gut Microbiome*.

## :dna::poop: Research 

I am interested in bacterial genomics, host-microbe interactions, and all things bacteriophage. My training is in molecular biology with a focus on DNA and RNA sequencing of microbiomes. While much of my time is spent at the bench, I am proficient in R, Bash, and many Linux-based sequencing analysis tools. 

In 2020, I co-authored a paper with [Dr. Alyssa Kent](https://alyssakent.weebly.com/) on the use of proximity ligation sequencing (commonly called Hi-C) to resolve mobile element carriage in the gut microbiomes of cancer patients. We found that Hi-C could detect the putative transfer of multi-drug resistance genes between commensal microbes and pathogens within individual patients through time. You can read more about it in Ilana's [*Behind the Paper*](https://go.nature.com/2FJPxqX) highlight or in my Twitter thread:

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Excited for the publication of my first paper as a PhD student in the Brito Lab <a href="https://twitter.com/ilanabrito123?ref_src=twsrc%5Etfw">@ilanabrito123</a> <a href="https://twitter.com/CornellBME?ref_src=twsrc%5Etfw">@CornellBME</a> <a href="https://twitter.com/NatureComms?ref_src=twsrc%5Etfw">@NatureComms</a>: &quot;Widespread transfer of mobile antibiotic resistance genes within individual gut microbiomes revealed through bacterial Hi-C&quot; <a href="https://t.co/4ZpBtGDA8m">https://t.co/4ZpBtGDA8m</a><br><br>1/10</p>&mdash; Albert Vill (@_AlbertVill) <a href="https://twitter.com/_AlbertVill/status/1300856406787207171?ref_src=twsrc%5Etfw">September 1, 2020</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

In 2022, I wrote a manuscript on the application of [precision run-on sequencing](https://doi.org/10.1038/nprot.2016.086) to assess nascent transcription in bacteria. When applied to human stool samples, PRO-seq captures transcription at metagenomic loci that are co-transcriptionally processed, including tRNAs and CRISPR arrays. The manuscript is available as a [preprint on bioRxiv](https://www.biorxiv.org/content/10.1101/2022.04.22.489220v1).

<blockquote class="twitter-tweet"><p lang="en" dir="ltr">My favorite result: PRO-seq captures active transcription across <a href="https://twitter.com/hashtag/CRISPR?src=hash&amp;ref_src=twsrc%5Etfw">#CRISPR</a> arrays, and patterns suggest that endonuclease processing of crRNAs is concurrent with pre-crRNA transcription <a href="https://t.co/vS1tbBI94x">pic.twitter.com/vS1tbBI94x</a></p>&mdash; Albert Vill (@_AlbertVill) <a href="https://twitter.com/_AlbertVill/status/1519751344273465344?ref_src=twsrc%5Etfw">April 28, 2022</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>

The Brito Lab is continuing to innovate in the microbiome mobile gene space with techniques like [OIL-PCR](https://doi.org/10.7554/eLife.66834) and [GCN models for HGT prediction](https://doi.org/10.1126/sciadv.abj5056). 

## :computer::bar_chart: Side Projects

I love R and I appreciate well-documented bioinformatics tools. When I write a handy piece of code, I like to optimize it for ease-of-use. Please see [my code & resources page](https://albertvill.com/projects) for descriptions of tools and scripts that I've developed as part of my ongoing research in metagenomics and bacterial genetics.

## :confetti_ball::partying_face: Fun Stuff

I'm a regular contributor on [Biology Stack Exchange](https://biology.stackexchange.com). Below are some of my favorite questions I've answered. Come ask more! 

<a href="https://biology.stackexchange.com/users/52196/acvill"><img src="https://stackexchange.com/users/flair/10848131.png" width="287" height="80" alt="profile for acvill on Stack Exchange, a network of free, community-driven Q&amp;A sites" title="profile for acvill on Stack Exchange, a network of free, community-driven Q&amp;A sites" /></a>

- [What are the limitations of 16S rRNA sequencing?](https://biology.stackexchange.com/a/104658/52196)
- [Does direction relative to origin of replication matter on small plasmids?](https://biology.stackexchange.com/a/100124/52196)
- [Which proteins are part of the most different protein complexes?](https://biology.stackexchange.com/a/92550/52196)
- [What fraction of bacterial proteins are membrane bound?](https://biology.stackexchange.com/a/98437/52196)
- [What are the reasons for using oligo-dT instead of oligo-U to isolate mRNAs?](https://biology.stackexchange.com/a/101635/52196)
- [Will all bacteria become resistant against all antibiotics in the long term?](https://biology.stackexchange.com/a/90122/52196)
- [What are the considerations for studying population genetics of bacterial pan-genomes?](https://biology.stackexchange.com/a/97169/52196)

<style>
p.caption {
  font-size: 0.9em;
  text-align: center;
  color: #7f1cad;
  margin-right: 5%;
  margin-left: 5%;
}
</style>

