---
title: "Code & Resources"
author: "Albert Vill"
date: ""
output: html_document
---
I upload standalone functions and short scripts to my [GitHub Gists page :computer:](https://gist.github.com/acvill).  

If you want a collaborator to help with data cleaning or visualization, please [contact me :email:](https://albertvill.com/contact/).

If you find any of my code or tools helpful, feel free to [buy me a coffee :coffee:](https://www.buymeacoffee.com/acvill).

---

## Projects

[![](https://github-readme-stats.vercel.app/api/pin/?username=acvill&repo=CRISPRviewR&bg_color=21436e&text_color=ededed&title_color=30ff53&hide_border=true&icon_color=ededed)](https://github.com/acvill/CRISPRviewR)

<details>
  <summary>Expand details...</summary>

[minCED](https://github.com/ctSkennerton/minced) (**Min**ing **C**RISPRs in **E**nvironmental **D**atasets) is a program used by [prokka](https://github.com/tseemann/prokka) and other pipelines for the annotation of CRISPRs in metagenomic assemblies. However, manually parsing the output of minCED for interesting CRISPRs is tedious, and comparing CRISPRs across samples is difficult because of differences in array assembly and orientation. CRISPRviewR attempts to address this issue by associating CRISPR arrays by shared consensus repeat sequences, permitting clustering and visualization of spacers between metagenomic assemblies. This package also includes a function to [fix truncated repeats](https://albertvill.com/posts/crt_repeats/). Check out the [CRISPRviewR vignette](https://albertvill.com/CRISPRviewR-vignette.html) for use cases and example plots.

<center>

![Example output from CRISPRviewR](/img/about/CRISPR23_oralmicrobiomes_numbers_bgwhite.svg "CRISPR arrays"){ width=90% }

</center>
</details>

[![](https://github-readme-stats.vercel.app/api/pin/?username=acvill&repo=bracken_plot&bg_color=21436e&text_color=ededed&title_color=30ff53&hide_border=true&icon_color=ededed)](https://github.com/acvill/bracken_plot)

<details>
  <summary>Expand details...</summary>

As a graduate student studying metagenomics, I frequently use Derrick Wood's [Kraken2](https://doi.org/10.1186/s13059-019-1891-0) software for accurate and lightning-fast classification of short reads. [Bracken](https://doi.org/10.7717/peerj-cs.104) is complimentary software that uses Kraken reports to produce genus- and species-level abundance estimates. Often, a first step in metagenomic analysis is examining the distribution of different organisms across samples. To that end, I've created a [simple shiny app](https://acvill.shinyapps.io/bracken_plot/) for quick, customizable plotting of taxonomic structures encoded in multi-sample Bracken reports. 

<center>

![Example output from the bracken_plot app showing genus-level relative abudance across oral microbiome samples. Don't mind the human read contamination!](/img/about/bracken_barplot.png "Bracken bar plot"){ width=75% }

</center>
</details>

[![](https://github-readme-stats.vercel.app/api/pin/?username=acvill&repo=nodeSeqs&bg_color=21436e&text_color=ededed&title_color=30ff53&hide_border=true&icon_color=ededed)](https://github.com/acvill/nodeSeqs)

<details>
  <summary>Expand details...</summary>

`nodeSeqs.sh` takes in a [**G**raphical **F**ragment **A**ssembly file](http://gfa-spec.github.io/GFA-spec/GFA1.html) and outputs sequences proximal to high-degree nodes. In the language of GFA, this script looks for *segments* that have a large number of *links*, and then filters those segments by *k*-mer coverage to preclude hits that are poorly assembled due to low coverage. The resulting .fasta can then be fed into your favorite sequence alignment software.  

For an example application of this script, please see my blog post: [***Lost at the crossroads: genes at the nodes of short-read assembly graphs***](https://albertvill.com/posts/nodeseqs_oral/).

<center>

![A partial Bandage graph for a plasmidome assembly shows segments connected by links](/img/about/bandage_sub.svg "Bandage Graph"){ width=90% }

</center>
</details>

[![](https://github-readme-stats.vercel.app/api/pin/?username=acvill&repo=findMotifs&bg_color=21436e&text_color=ededed&title_color=30ff53&hide_border=true&icon_color=ededed)](https://github.com/acvill/findMotifs)

<details>
  <summary>Expand details...</summary>
  
To understand the genetic context and biological relevance of DNA motifs, it's often valuable to analyze their distribution and sequence conservation. Many programs exist for both the identification of novel motifs and the scoring of known motifs, though few of these tools work out-of-the-box for highly fragmented bacterial genomes. To address this, I've written findMotifs.R -- an easy-to-implement R script to find and score short sequences using position-frequency matrices (PFMs). Given a set of contigs and a list of PFMs, this script returns an easy-to-parse table containing the sequence, score, and strand-wise position of each match above a user-defined threshold. findMotifs.R requires R version 4.0 and three packages: Biostrings, optparse, and stringr.

<center>

![findmotifs.R is invoked from the command line with user inputs. Progress for each motif prints to the console.](/img/about/findmotifs2.PNG "command line screenshot of findMotifs in action"){ width=90% }

</center>
</details>

<style>
p.caption {
  font-size: 0.9em;
  text-align: center;
  color: #7f1cad;
  margin-right: 10%;
  margin-left: 10%;
}
details > summary {
  font-size: 1.1em;
  padding: 0px;
  background-color: #FFFFFF;
  border: none;
  box-shadow: 0px 0px 0px #bbbbbb;
  cursor: pointer;
  margin-left: 5%;
  color: #21436e;
  border-radius: 5px;
}
</style>

