<!DOCTYPE html>
<html lang="en">
    
    


    <head>
    <link href="https://gmpg.org/xfn/11" rel="profile">
    <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<!-- Enable responsiveness on mobile devices -->
<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
<meta name="generator" content="Hugo 0.105.0">

    
    
    

<title>Lost at the crossroads: genes at the nodes of short-read assembly graphs • Albert Vill</title>


<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Lost at the crossroads: genes at the nodes of short-read assembly graphs"/>
<meta name="twitter:description" content="


A brief functional analysis of the sequence content of high-degree segments in metaSPAdes graphs.
"/>

<meta property="og:title" content="Lost at the crossroads: genes at the nodes of short-read assembly graphs" />
<meta property="og:description" content="


A brief functional analysis of the sequence content of high-degree segments in metaSPAdes graphs.
" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://acvill.github.io/posts/nodeseqs_oral/" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2022-03-05T00:00:00+00:00" />
<meta property="article:modified_time" content="2022-03-05T00:00:00+00:00" /><meta property="og:site_name" content="Title" />



    


<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">








<link rel="stylesheet" href="/scss/hyde-hyde.83a929a933962c7643b655554c2e1b7cb32e23c25a2d7262666d4ac0574b9e7e.css" integrity="sha256-g6kpqTOWLHZDtlVVTC4bfLMuI8JaLXJiZm1KwFdLnn4=">


<link rel="stylesheet" href="/scss/print.2744dcbf8a0b2e74f8a50e4b34e5f441be7cf93cc7de27029121c6a09f9e77bc.css" integrity="sha256-J0Tcv4oLLnT4pQ5LNOX0Qb58&#43;TzH3icCkSHGoJ&#43;ed7w=" media="print">



    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
    <script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
    <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
    <!-- Icons -->
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/img/apple-touch-icon.png">
    <link rel="shortcut icon" href="/img/favicon-32x32.png">
    
    

    
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"> </script>
</head>


    <body class=" ">
    
<div class="sidebar">
  <div class="container ">
    <div class="sidebar-about">
      <span class="site__title">
        <a href="https://acvill.github.io/">
        
          Albert Vill
        
        </a>
      </span>
      
        
        
        
        <div class="author-image">
          <img src="https://acvill.github.io/img/about/july2021_cropped1.png" alt="Author Image" class="img--circle img--headshot element--center">
        </div>
        
      
      
      
      <p class="site__description">
        
      </p>
    </div>
    <div class="collapsible-menu">
      <input type="checkbox" id="menuToggle">
      <label for="menuToggle">Albert Vill</label>
      <div class="menu-content">
        <div>
	<ul class="sidebar-nav">
		 
		 
			 
				<li>
					<a href="/about">
						<span>About</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/cv">
						<span>CV</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/posts/">
						<span>Blog</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/projects">
						<span>Code &amp; Resources</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/contact">
						<span>Contact Me</span>
					</a>
				</li>
			 
		
	</ul>
</div>

        <section class="social">
  
	<a href="https://biology.stackexchange.com/users/52196/acvill" rel="me"><i class="fab fa-stack-exchange fa-lg" aria-hidden="true"></i></a>
	
	
	<a href="https://twitter.com/_AlbertVill" rel="me"><i class="fab fa-twitter fa-lg" aria-hidden="true"></i></a>
	
	
	
	
	<a href="https://github.com/acvill" rel="me"><i class="fab fa-github fa-lg" aria-hidden="true"></i></a>
	
	
	
	
	
	
	
	<a href="https://linkedin.com/in/albert-vill-a457821a1" rel="me"><i class="fab fa-linkedin fa-lg" aria-hidden="true"></i></a>
	
	
	<a href="https://scholar.google.com/citations?user=j9imsbgAAAAJ" rel="me"><i class="fas fa-graduation-cap fa-lg" aria-hidden="true"></i></a>
	
	
	
	
	
	
	
	
	
	
	
	
</section>



      </div>
    </div>
    
<div class="copyright">
  &copy; 2020 - 2023 A.C. Vill
  
  <br>
    <a href="https://creativecommons.org/licenses/by-sa/4.0">CC BY-SA 4.0</a>
  
</div>



  </div>
</div>

        <div class="content container">
            
    
<article>
  <header>
    <h1>Lost at the crossroads: genes at the nodes of short-read assembly graphs</h1>
    
    
<div class="post__meta">
    
    
      <i class="fas fa-calendar-alt"></i> 5 Mar 2022
    
    
    
    
    
      
      
          <br/>
           <i class="fas fa-tags"></i>
          
          <a class="badge badge-tag" href="/tags/metagenomics">metagenomics</a>
         
      
          <a class="badge badge-tag" href="/tags/bioinformatics">bioinformatics</a>
         
      
          <a class="badge badge-tag" href="/tags/r">r</a>
        
      
    
    
    <br/>
    
</div>


  </header>
  
  
  <div class="post">
    </p>
<center>
<p><em>This post is a continuation of my question on <a href="https://bioinformatics.stackexchange.com/q/18476/3967">Bioinformatics Stack Exchange</a>.</em></p>
<div class="figure">
<img src="/img/posts/pspades_hairball_v4.png" title="Bandage Graph" style="width:100.0%" alt="" />
<p class="caption">A subset of a <a href="https://github.com/rrwick/Bandage">Bandage graph</a> for a plasmidSPAdes assembly shows a hairball of segments connected by links</p>
</div>
</center>
<p>I often use <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5411777/">metaSPAdes</a> to assemble short reads from human microbiomes. My simplified understanding of short-read de Bruijn graph assemblers is that they fail where ambiguous paths cannot be resolved. While it can be said that these points of failure may be due to certain <strong>structural</strong> features of the underlying sequences – long repeats or conserved sequences found at multiple loci – I’m interested in the <strong>functional</strong> classification of sequences that are proximal to nodes on assembly graphs. Put another way, what kinds of genes are poorly represented in metagenomes because they are poorly assembled? Naively, I expect mobile elements like transposons and integrons to be in such a list of genes, since they are often flanked by repeated sequences and can be present in many copies across multiple genomic contexts, though I want to take an empirical approach.</p>
<p>To that end, I wrote a script to parse assembly graphs in <a href="https://web.archive.org/web/20211209195521/http://gfa-spec.github.io/GFA-spec/GFA1.html"><strong>G</strong>raphical <strong>F</strong>ragment <strong>A</strong>ssembly</a> format. Given a GFA file, <code>nodeSeqs.sh</code> will (1) extract segments with at least a minimum number of links, (2) filter those segments given a specified minimum coverage, and (3) trim segment sequences with respect to the orientation of their links. Click the card below to check out the Github repo.</p>
<center>
<p><a href="https://github.com/acvill/nodeSeqs"><img src="https://github-readme-stats.vercel.app/api/pin/?username=acvill&amp;repo=nodeSeqs&amp;bg_color=21436e&amp;text_color=ededed&amp;title_color=30ff53&amp;hide_border=true&amp;icon_color=ededed" /></a></p>
</center>
<p>I applied nodeSeqs to six GFA files, each recovered from a human oral microbiome assembly created using ~50 million paired-end reads and default metaSPAdes parameters.</p>
<details>
<p><summary>View Code: run nodeSeqs.sh</summary></p>
<pre class="bash"><code>./nodeSeqs.sh \
  -g assembly_graph_with_scaffolds.gfa \
  -d 5 \
  -c 200 \
  -m 25</code></pre>
</details>
<p>Looking at the resulting .fasta files, some of the sequences consist of long stretches of mono- and di-nucleotide repeats, but most of the extracted segments lack any discernible repeat structure. To get at the putative functional roles of these sequences, I ran each sequence set through <a href="https://github.com/eggnogdb/eggnog-mapper">eggNOG-mapper</a>, which is built to annotate novel sequences using clustered orthologous groups (COGs) of proteins.</p>
<details>
<p><summary>View Code: install and run eggNOG-mapper</summary></p>
<pre class="bash"><code>source /home/miniconda3/bin/activate
conda create --name eggnog-mapper
conda install --channel bioconda eggnog-mapper
conda activate eggnog-mapper

download_eggnog_data.py

# emapper-2.1.7
# Installed eggNOG DB version: 5.0.2
# Diamond version found: diamond version 2.0.14
# MMseqs2 version found: 13.45111

emapper.py \
  -i s1_segments.fasta \
  --itype CDS \
  -m diamond \
  -o s1 \
  --sensmode ultra-sensitive \
  --cpu 8</code></pre>
</details>
<p>With COG annotations, we can then ask which functions are highly represented among high-degree segments. Plotting the top five functions as proportions of the total number of segments for each sample, we get some interesting results…</p>
<details>
<p><summary>View Code: process data and plot results</summary></p>
<pre class="bash"><code># clean up annotation headers
sed -i &#39;s/#query/query/g&#39; s*.emapper.annotations

# get total number of segments for each sample, for normalization
for file in ./s*_segments.fasta; do
  num=$(grep -v &quot;&gt;&quot; $file | wc -l)
  echo ${file} | \
  awk -F$&#39;/&#39; &#39;{print $2}&#39; | \
  sed &quot;s/^/${num}\t/g&quot;
done &gt; segment_counts.txt</code></pre>
<pre class="r"><code># process and merge annotation data in the tidyverse
library(tidyverse)
setwd(&quot;/workdir/users/acv46/nodeSeqs&quot;)

segcounts &lt;- read_tsv(&quot;segment_counts.txt&quot;, col_names = c(&quot;count&quot;,&quot;sample&quot;))

read_emap &lt;- function(file, segcounts) {
  # file is a list of emapper annotation files
  # segcounts is total segment counts per sample, for normalization
  read_tsv(file = file, comment = &quot;##&quot;, col_names = T) %&gt;%
    mutate(sample = str_remove(file, &quot;.emapper.annotations&quot;)) %&gt;%
    mutate(totseg = segcounts[which(segcounts$sample == str_remove(file, &quot;.emapper.annotations&quot;)),]$count)
}

s1 &lt;- read_emap(&quot;s1.emapper.annotations&quot;, segcounts)
s2 &lt;- read_emap(&quot;s2.emapper.annotations&quot;, segcounts)
s3 &lt;- read_emap(&quot;s3.emapper.annotations&quot;, segcounts)
s4 &lt;- read_emap(&quot;s4.emapper.annotations&quot;, segcounts)
s5 &lt;- read_emap(&quot;s5.emapper.annotations&quot;, segcounts)
s6 &lt;- read_emap(&quot;s6.emapper.annotations&quot;, segcounts)

merge &lt;- bind_rows(s1,s2,s3,s4,s5,s6) %&gt;%
  mutate(COG = sub(&quot;@1.*&quot;, &quot;&quot;, eggNOG_OGs)) %&gt;%
  filter(grepl(&#39;COG&#39;, COG)) %&gt;%
  select(sample, COG, totseg) %&gt;%
  group_by(sample, COG) %&gt;% mutate(prop = 100*n()/totseg) %&gt;%
  ungroup %&gt;% distinct %&gt;% select(-totseg) %&gt;%
  complete(sample, COG, fill = list(prop = 0)) %&gt;%
  group_by(COG) %&gt;% mutate(avg = median(prop)) %&gt;%
  ungroup %&gt;% arrange(desc(avg))</code></pre>
<pre class="r"><code># get top functions by median across samples
top5 &lt;- merge %&gt;% select(COG, avg) %&gt;%
  unique %&gt;% slice(1:5) %&gt;% select(COG) %&gt;% as_vector</code></pre>
<pre class="r"><code># plot bar graph
pal &lt;- c(&#39;#33658a&#39;, &#39;#f6ae2d&#39;, &#39;#86bbd8&#39;, &#39;#f26419&#39;, &#39;#2f4858&#39;, &#39;#f71735&#39;)
ggplot(data = merge %&gt;% filter(COG %in% top5)) +
  geom_bar(mapping = aes(x = factor(COG, level = top5), y = prop, fill = sample),
           stat = &quot;identity&quot;, position = &quot;dodge&quot;, width = 0.75) +
  theme_bw() +
  scale_fill_manual(values = pal) +
  ylab(&quot;percent of segments&quot;) +
  theme(axis.title.x = element_blank(),
        axis.title.y = element_text(size = 10),
        axis.text.x = element_text(size = 8, angle = 0,
                                   vjust = 0.5, color = &quot;black&quot;),
        axis.text.y = element_text(size = 8, color = &quot;black&quot;),
        legend.position = c(0.8, 0.7),
        legend.background = element_rect(fill=&quot;grey90&quot;,
                                  size = 0.5, linetype = &quot;solid&quot;, 
                                  colour = &quot;black&quot;),
        legend.text = element_text(size = 8, color = &quot;black&quot;),
        legend.title = element_blank(),
        legend.key.size = unit(3, &#39;mm&#39;),
        legend.key = element_rect(color = &quot;black&quot;), 
        panel.border = element_rect(color = &quot;black&quot;, size = 1.5),
        axis.line.y = element_blank(),
        axis.line.x = element_blank(),
        panel.grid.minor = element_blank(),
        axis.line = element_line(color = &quot;black&quot;)) +
  guides(fill = guide_legend(nrow = 3, byrow = TRUE))</code></pre>
</details>
<center>
<div class="figure">
<img src="/img/posts/nodeSeqs_top5_4w2h_oral_v4.png" title="bar graph" style="width:100.0%" alt="" />
<p class="caption">Bar graph showing the proportion of high-degree segments from six human oral microbiome assembly graphs that encode proteins which match specific COGs. The 5 COGs with the highest median percentage across samples are shown.</p>
</div>
</center>
<p style="padding: 10px; border: 2px solid #21436e;">
<a href="https://www.ncbi.nlm.nih.gov/Structure/cdd/COG3415"><strong>COG3415</strong></a>: Transposase<br />
<a href="https://www.ncbi.nlm.nih.gov/Structure/cdd/COG1961"><strong>COG1961</strong></a>: Site-specific DNA recombinase<br />
<a href="https://www.ncbi.nlm.nih.gov/Structure/cdd/COG3328"><strong>COG3328</strong></a>: Transposase (or an inactivated derivative)<br />
<a href="https://www.ncbi.nlm.nih.gov/Structure/cdd/COG0370"><strong>COG0370</strong></a>: Fe2+ transport system protein B<br />
<a href="https://www.ncbi.nlm.nih.gov/Structure/cdd/COG1073"><strong>COG1073</strong></a>: Fermentation-respiration switch protein FrsA
</p>
<p>Transposases and recombinases, both common components of mobile elements, are overrepresented at the high-degree junctions of metaSPAdes assembly graphs. This is unsurprising, given the ubiquity of these elements and their sequence conservation across phylogenetic space. The consequence of this is that chromosome-integrated mobile elements are often underrepresented in metagenome-assembled genomes (MAGs). It’s possible that these assemblies could be made better by <a href="https://web.archive.org/web/20211209202743/https://github.com/rrwick/Bandage/wiki/Effect-of-kmer-size">tweaking the <em>k</em>-mer size</a>, but the greatest potential for improving the contiguity and completeness of MAGs lies in alternative sequencing methods. Long-read sequencing of microbiomes results in <a href="https://doi.org/10.1186/s12864-021-07607-0">more circular contigs</a> and better resolution on the taxonomic associations of the so-called ‘mobilome’. Likewise, proximity ligation sequencing can identify the <em>in situ</em> hosts of <a href="https://doi.org/10.1038/s41467-020-18164-7">plasmids</a> and <a href="https://doi.org/10.1101/2021.06.14.448389">phage</a>, circumventing the need for association by contiguity.</p>
<style>
p.caption {
  font-size: 0.9em;
  text-align: center;
  color: #7f1cad;
  margin-right: 10%;
  margin-left: 10%;
}
details > summary {
  font-size: 1.0em;
  padding: 10px;
  background-color: #21436e;
  border: none;
  box-shadow: 0px 0px 0px #bbbbbb;
  cursor: pointer;
  margin-left: 5%;
  margin-right: 5%;
  margin-top: 0%;
  margin-bottom: 0%;
  color: #FFFFFF;
  border-radius: 5px;
}
</style>

  </div>
  


  

  
    


</article>


        </div>
        
    

  
  
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'G-JKQWQF4NBY', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>


<script defer src="https://use.fontawesome.com/releases/v5.12.1/js/all.js" integrity="sha384-ZbbbT1gw3joYkKRqh0kWyRp32UAvdqkpbLedQJSlnI8iLQcFVxaGyrOgOJiDQTTR" crossorigin="anonymous"></script>


    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.13.1/highlight.min.js"></script>
        
    <script type="text/javascript">
        
        hljs.initHighlightingOnLoad();
    </script>
    



    



    </body>
</html>
