<!DOCTYPE html>
<html lang="en">
    
    


    <head>
    <link href="https://gmpg.org/xfn/11" rel="profile">
    <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<!-- Enable responsiveness on mobile devices -->
<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1">
<meta name="generator" content="Hugo 0.105.0">

    
    
    

<title>Parsing GTF files in the tidyverse • Albert Vill</title>


<meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="Parsing GTF files in the tidyverse"/>
<meta name="twitter:description" content="


An R &#43; tidyverse approach for reading GTF annotations.
"/>

<meta property="og:title" content="Parsing GTF files in the tidyverse" />
<meta property="og:description" content="


An R &#43; tidyverse approach for reading GTF annotations.
" />
<meta property="og:type" content="article" />
<meta property="og:url" content="https://acvill.github.io/posts/gtf_tidy/" /><meta property="article:section" content="posts" />
<meta property="article:published_time" content="2021-11-09T00:00:00+00:00" />
<meta property="article:modified_time" content="2021-11-09T00:00:00+00:00" /><meta property="og:site_name" content="Title" />



    


<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">








<link rel="stylesheet" href="/scss/hyde-hyde.83a929a933962c7643b655554c2e1b7cb32e23c25a2d7262666d4ac0574b9e7e.css" integrity="sha256-g6kpqTOWLHZDtlVVTC4bfLMuI8JaLXJiZm1KwFdLnn4=">


<link rel="stylesheet" href="/scss/print.2744dcbf8a0b2e74f8a50e4b34e5f441be7cf93cc7de27029121c6a09f9e77bc.css" integrity="sha256-J0Tcv4oLLnT4pQ5LNOX0Qb58&#43;TzH3icCkSHGoJ&#43;ed7w=" media="print">



    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
    <script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
    <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->
    <!-- Icons -->
    <link rel="apple-touch-icon-precomposed" sizes="144x144" href="/img/apple-touch-icon.png">
    <link rel="shortcut icon" href="/img/favicon-32x32.png">
    
    

    
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"> </script>
    
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-JKQWQF4NBY"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-JKQWQF4NBY');
    </script>
</head>


    <body class=" ">
    
<div class="sidebar">
  <div class="container ">
    <div class="sidebar-about">
      <span class="site__title">
        <a href="https://acvill.github.io/">
        
          Albert Vill
        
        </a>
      </span>
      
        
        
        
        <div class="author-image">
          <img src="https://acvill.github.io/img/about/picofme_7.png" alt="Author Image" class="img--circle img--headshot element--center">
        </div>
        
      
      
      
      <p class="site__description">
        
      </p>
    </div>
    <div class="collapsible-menu">
      <input type="checkbox" id="menuToggle">
      <label for="menuToggle">Albert Vill</label>
      <div class="menu-content">
        <div>
	<ul class="sidebar-nav">
		 
		 
			 
				<li>
					<a href="/about">
						<span>About</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/cv">
						<span>CV</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/posts/">
						<span>Blog</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/projects">
						<span>Code &amp; Resources</span>
					</a>
				</li>
			 
		 
			 
				<li>
					<a href="/contact">
						<span>Contact Me</span>
					</a>
				</li>
			 
		
	</ul>
</div>

        <section class="social">
  
	<a href="https://biology.stackexchange.com/users/52196/acvill" rel="me"><i class="fab fa-stack-exchange fa-lg" aria-hidden="true"></i></a>
	
	
	<a href="https://twitter.com/_AlbertVill" rel="me"><i class="fab fa-twitter fa-lg" aria-hidden="true"></i></a>
	
	
	
	
	<a href="https://github.com/acvill" rel="me"><i class="fab fa-github fa-lg" aria-hidden="true"></i></a>
	
	
	
	
	
	
	
	<a href="https://linkedin.com/in/albert-vill-a457821a1" rel="me"><i class="fab fa-linkedin fa-lg" aria-hidden="true"></i></a>
	
	
	<a href="https://scholar.google.com/citations?user=j9imsbgAAAAJ" rel="me"><i class="fas fa-graduation-cap fa-lg" aria-hidden="true"></i></a>
	
	
	
	
	
	
	
	
	
	
	
	
</section>



      </div>
    </div>
    
<div class="copyright">
  &copy; 2020 - 2023 A.C. Vill
  
  <br>
    <a href="https://creativecommons.org/licenses/by-sa/4.0">CC BY-SA 4.0</a>
  
</div>



  </div>
</div>

        <div class="content container">
            
    
<article>
  <header>
    <h1>Parsing GTF files in the tidyverse</h1>
    
    
<div class="post__meta">
    
    
      <i class="fas fa-calendar-alt"></i> 9 Nov 2021
    
    
    
    
    
      
      
          <br/>
           <i class="fas fa-tags"></i>
          
          <a class="badge badge-tag" href="/tags/metagenomics">metagenomics</a>
         
      
          <a class="badge badge-tag" href="/tags/bioinformatics">bioinformatics</a>
         
      
          <a class="badge badge-tag" href="/tags/r">r</a>
        
      
    
    
    <br/>
    
</div>


  </header>
  
  
  <div class="post">
    </p>
<p>Like many other people working with sequencing data in R, I rely on different packages to do different things. For reading annotations in GTF and GFF formats, there are a few well-regarded choices to get the job done: refGenome and rtracklayer (see <a href="https://davetang.org/muse/2017/08/04/read-gtf-file-r/">Dave Tang’s blog</a> for a description of each). Unfortunately, refGenome has not been updated since 2019 and <a href="https://CRAN.R-project.org/package=refGenome">has since been removed from the CRAN repository</a>. As for rtracklayer, it is updated regularly as part of the Bioconductor suite of tools, though I’ve found <code>import()</code> can be finicky if your annotation file contains unexpected characters in certain places. Moreover, R’s opaque error reporting can make it <a href="https://www.biostars.org/p/488193/#488193">hard to diagnose</a> whether an error is caused by rtracklayer or one of its dependencies.</p>
<center>
<p><img src="/img/posts/stringr_dplyr.png" title="stringr logo" style="width:70.0%" /></p>
</center>
<p>I love the tidyverse. Though it has a <a href="https://stackoverflow.com/a/61393520/7976890">reputation for instability</a>, I find the consistent grammar and interoperability of its packages make data cleaning a simple and transparent process. And when breaking changes are implemented, the tidyverse’s <a href="https://lifecycle.r-lib.org/articles/stages.html">meticulous lifecycle documentation</a> means that deprecated code is easy to track down and change. So, I sought a tidyverse approach to read genome annotations. Below is my code to read in a GTF file as a tibble using functions from stringr and dplyr.</p>
<script type="application/javascript" src="https://gist.github.com/acvill/03343034392cff158d2369483ed8935f.js"></script>
<p>I would like to credit Stack Overflow user akrun for <a href="https://stackoverflow.com/a/69901344/7976890">providing the regex for attribute string parsing</a>.</p>
<p>This above function correctly parses the <code>attribute</code> column for files in which the number and types of attributes are inconsistent between features. It also handles instances where the attribute field separator (<code>;</code>) is found in quoted attribute strings. Compare this to <code>tidy_gff()</code> from <a href="https://rdrr.io/github/vsbuffalo/annotatr/man/tidy_gff.html">Vince Buffalo’s annotatr package</a>, which parses GFF files using a similar tidy scheme. Digging into the <a href="https://rdrr.io/github/vsbuffalo/annotatr/src/R/annotatr.r">source code</a>, however, we see that the function only extracts a fixed subset of attribute types: <code>ID</code>, <code>Parent</code>, and <code>parent_type</code>.</p>
<pre class="r"><code>extract_attr &lt;- function(x, key)  {
  has_key &lt;- regexpr(key, x)
  value &lt;- gsub(sprintf(&#39;.*%s=([^;]+).*&#39;, key), &#39;\\1&#39;, x)
  value[has_key == -1] &lt;- NA
  out &lt;- strsplit(value, &#39;,&#39;)
  multi_value &lt;- any(sapply(out, length) &gt; 1)
  if (!multi_value) return(unlist(out))
  out
}

tidy_gff &lt;- function(x, chroms=NULL,
                     features=c(&#39;five_prime_UTR&#39;, &#39;three_prime_UTR&#39;, 
                                &#39;exon&#39;, &#39;intron&#39;, &#39;gene&#39;, &#39;CDS&#39;)) {
  out &lt;- dplyr::select(dplyr::mutate(dplyr::filter(x, feature %in% features), 
                id=extract_attr(attr, &#39;ID&#39;),
                parent=extract_attr(attr, &#39;Parent&#39;),
                parent_type=extract_attr(attr, &#39;parent_type&#39;)),
        chrom, start, end, strand, feature, id, parent, parent_type)
  if (!is.null(chroms)) {
    out[out$chrom %in% chroms, ]
  }
  out
}</code></pre>
<p>For most applications, this is sufficient and fast. However, metagenomic annotations often contain exotic attribute types, as is the case with files output by programs that aggregate features from multiple sources like <a href="https://github.com/tseemann/prokka">Torsten Seemann’s prokka</a>. In such analyses, you may want the full attribute set, and <code>read_gtf()</code> is here to help.</p>

  </div>
  


  

  
    


</article>


        </div>
        
    

  
  
<script type="application/javascript">
var doNotTrack = false;
if (!doNotTrack) {
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', 'G-JKQWQF4NBY', 'auto');
	
	ga('send', 'pageview');
}
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>


<script defer src="https://use.fontawesome.com/releases/v5.12.1/js/all.js" integrity="sha384-ZbbbT1gw3joYkKRqh0kWyRp32UAvdqkpbLedQJSlnI8iLQcFVxaGyrOgOJiDQTTR" crossorigin="anonymous"></script>


    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.13.1/highlight.min.js"></script>
        
    <script type="text/javascript">
        
        hljs.initHighlightingOnLoad();
    </script>
    



    



    </body>
</html>
